### PR TITLE
xGov-26 - ASA Stats Offer - Algorand blockchain ASA/NFT creators' channel

### PR DESCRIPTION
**Pull request by ipaleka**
ASA Stats offer represents a proposal to exchange two or more ASA between an authenticated Algorand address owner and the owner of any other Algorand address. An authenticated user is a person who has proven ownership of one or multiple Algorand addresses.

After both participants approve an offer, the assets defined by the offer are transferred to the recipients' addresses through the proxy escrows. Ultimately, two identical ASA Stats Offer NFTs are created and allocated to the participants' addresses. The owners of those NFTs are allowed to repeat the same offer an unlimited number of times free of charge.

### CODE REVIEW

File: Proposals/xgov-26.md [hunk 1]
```diff
@@ -0,0 +1,211 @@
+---
+title: ASA Stats Offer - Algorand blockchain ASA/NFT creators' channel
+author: Ivica Paleka (@ipaleka), Marcin Zawiejski (@dragmz), Eduard Ravnic (@kerrilija)
+company_name: ASA Stats
+category: Tools
+focus_area: User Onboarding
+open_source: Yes
+amount_requested: 250000
+status: Draft
```
Comment (SudoWeezy) [inline Proposals/xgov-26.md:9]: Remember to change this to "Final" if you want your proposal to go through the next voting session.

File: Proposals/xgov-26.md [hunk 2]
```diff
@@ -0,0 +1,211 @@
+---
+title: ASA Stats Offer - Algorand blockchain ASA/NFT creators' channel
+author: Ivica Paleka (@ipaleka), Marcin Zawiejski (@dragmz), Eduard Ravnic (@kerrilija)
+company_name: ASA Stats
+category: Tools
+focus_area: User Onboarding
+open_source: Yes
+amount_requested: 250000
```
Comment (SudoWeezy) [inline Proposals/xgov-26.md:8]: If your project requires multiple milestones, split them into multiple proposals. See: https://github.com/algorandfoundation/ARCs/pull/209

Comment (ipaleka) [reply Proposals/xgov-26.md:8]: > If your project requires multiple milestones, split them into multiple proposals. See: [algorandfoundation/ARCs#209](https://github.com/algorandfoundation/ARCs/pull/209)

Afaic that's contrary to the guideline from the [provided template](https://github.com/algorandfoundation/ARCs/blob/main/assets/arc-0034/TemplateForm.md#roadmap):

Comment (ipaleka) [reply Proposals/xgov-26.md:8]: So here are first three milestones from the proposal:

-----

### Milestone 1: Proof of concept for the user authentication

Date: 31/10/2023  
Amount: 2,000  
Description: Research existing technologies and possible ways for creating a transaction that will prove the user's ownership of an Algorand address. Create a working prototype.

### Milestone 2: Proof of concept for the atomic transfers to and from the escrows

Date: 31/12/2023  
Amount: 8,000  
Description: Research existing technologies and possible ways for creating the escrows and signing the atomic transfers from and to them from the user accounts. Define limits for the number of different ASA that can be involved in an offer and propose the optimal and maximum numbers. Create a working prototype.

### Milestone 3: Proof of concept for the non-users' notification transaction creation

Date: 31/10/2023  
Amount: 2,000  
Description: If the target address - to which the offer refers - hasn't been connected to an ASA Stats user then a transaction should be sent to the target address with the minimal amount of ALGO needed to trigger a notification on major wallet apps. Research, develop, and deploy a prototype.

-----

So should I rewrite proposal in a way that every milestone consists of work that lasts for about 3 (or multiplier) months?

Like:

-----
```

---
title: ASA Stats Offer - Algorand blockchain ASA/NFT creators' channel
author: Ivica Paleka (@ipaleka), Marcin Zawiejski (@dragmz), Eduard Ravnic (@kerrilija)
company_name: ASA Stats
category: Tools
focus_area: User Onboarding
open_source: Yes
amount_requested: 12000
status: Final
---
```
...

### Milestone 1: Proof of concepts

Date: 3 months  
Amount: 12,000  
Description: Research existing technologies and possible ways for:
 - define a process for creating a transaction that will prove the user's ownership of an Algorand address. Create a working prototype.
 - define a process for creating the escrows and signing the atomic transfers from and to them from the user accounts. Define limits for the number of different ASA that can be involved in an offer and propose the optimal and maximum numbers.
 - define a process for creating a transaction should be sent to the target address with the minimal amount of ALGO needed to trigger a notification on major wallet apps

### Milestone 2: ...
Date: 3 months  
Amount: 25,000  

### Milestone 3: ...
Date: 6 months  
Amount: 58,000  

### Milestone 4: ...

-----

**Comment by nonfungibleab** [comment]
Hi @ipaleka 
Would you please update your proposal based on the latest criteria for milestone-based proposals. Ping me when done, so I can add it to the forum for discussion.

**Details are below, but in short, you must submit a different grant for every milestone. Submit a grant for the first milestone, if approved you get paid, and work commences. Once the milestone is complete, you submit a grant for milestone 2 with proof of milestone 1 completion. And so forth for other milestones until the project is finished.**

https://arc.algorand.foundation/ARCs/arc-0034
Submit a proposal
In order to submit a proposals, a proposer needs to create a pull request on the following repository: [xGov Proposals](https://github.com/algorandfoundation/xGov).

The tutorial can be found here: [xGov proposal submission Tutorial](https://www.algorand.foundation/xgov-how-to-submit-grant-proposals).

Proposals MUST:

Follow the [template form provided](https://arc.algorand.foundation/assets/arc-0034/TemplateForm).

- Follow the rules of the xGov Proposals Repository.
- Have the status Final before the end of the temperature check.
- Milestone-based grants must submit a proposal for one milestone at a time.
- Milestones need to follow the governance periods cycle. With the current 3-months cycle, a milestone could be 3-months, 6 months, 9 months etc.
- The initial proposal must include display all milestones with clear deliverables **and the amount requested must match the 1st milestone.**
- Once a milestone is achieved, the team must submit a proposal for the next milestone, including ways for xGovs to verify that the milestone is completed.

**Comment by ipaleka** [comment]
> Hi @ipaleka Would you please update your proposal based on the latest criteria for milestone-based proposals. Ping me when done, so I can add it to the forum for discussion.

Hi @nonfungibleab !

The proposal is updated, please check. :) I'm not sure of the format for a milestone date, I set it to duration ("3 months"). If that's ok I'm going to set the status to Final.

**Comment by ShaperOfEntropy** [comment]
An interesting tool for the ecosystem. Good cost-effort ratio.

What I'm missing in the proposal is similar to your other proposal:

1. Analysis of the competition and existing solutions in the ecosystem and beyond. How does your project compare to them? How used are they?
2. Who is your target customer? Will this be B2C solution or B2B, where you will be integrating with existing marketplaces, creators, etc.?
3. How will xGovs be able to verify the deliverables of each milestone? Will POCs be publicly available to test? If so, how (e.g. GUI, CLI, or automated tests)? Will outcome of research and design stage be published? How would you define the public discussion with NFT creators successful?
4. How will the grant be spent? Will ALGO be sold on the market? If yes, using DEXs or CEXs? Or will a loan be taken against it in DeFi/CeFi? Or will project team members be paid directly in ALGO? Or will it go to project treasury? Or for buyback of your tokens?

Since you have multiple proposals, it raises the question whether you would be able to deliver on both of them if they both get approved, especially since you say the core functionality of ASA Stats has the highest priority.

**Comment by ipaleka** [comment]
> An interesting tool for the ecosystem. Good cost-effort ratio.

Thank you for this and also thank you for asking these questions!

> What I'm missing in the proposal is similar to your other proposal:

> 1. Analysis of the competition and existing solutions in the ecosystem and beyond. How does your project compare to them? How used are they?

No competition for the ASA Stats Offer exists. [Here's another answer](https://forum.algorand.org/t/xgov-26-asa-stats-offer-algorand-blockchain-asa-nft-creators-channel/9903/4?u=ipaleka) on Algorand's forum, but in short, EXA Swap, Atomixwap and/or Swapper are on-chain tools with predefined set of assets that'd be exchanged.

ASA Stats Offer is a completely different off-chain tool that brings *negotiation* and *communication* levels into the process, it brings evaluation of all the involved assets in **real-time**, and, of course, it allows changing of all the offer's aspects in real-time visible to both participants.

Therefore their usage couldn't represent any reference for ASA Stats Offer usage.

> Who is your target customer? Will this be B2C solution or B2B, where you will be integrating with existing marketplaces, creators, etc.?

Every Algorand user and beyond - as there's only one label that can be set for "focus_area" we've used "User Onboarding".

On our website and mobile apps there will be a user's offers page, a specific offer page visible to both participants, and probably a selectable user widget that will allow users to click the "Add to offer" button while they are browsing some address page.

> How will xGovs be able to verify the deliverables of each milestone? Will POCs be publicly available to test? If so, how (e.g. GUI, CLI, or automated tests)? Will outcome of research and design stage be published? 

We are probably one of [the most transparent projects](https://www.asastats.com/tokenomics/) in the ecosystem, so it is implied that we publish everything as specified in our milestones, either in the form of source code or documentation.

> How would you define the public discussion with NFT creators successful?

We won't set any quality marks on it. We'd conduct it by trying to reach as many different types of NFT creators and publishers, as well as with those that have never created and published anything, and then we'd try to respect individual suggestions and/or requests if that's possible. In short, just the way [we always do](https://github.com/asastats/channel/wiki/Discussions).

> How will the grant be spent? Will ALGO be sold on the market? If yes, using DEXs or CEXs? Or will a loan be taken against it in DeFi/CeFi? Or will project team members be paid directly in ALGO? Or will it go to project treasury? Or for buyback of your tokens?

Dunno. We did get around US$24,000 in ALGO in a private presale from our community members back in Autumn 2021., but we spend around 10-15% of that money in the last 20 months while the rest has been eaten by the bear or provided to liquidity.

It is forbidden by the internal policy that we trade our token, the only that is allowed is to sell the Development Pool's and Administration and Costs Pool's ASASTATS on the open market but we simply haven't played that card yet in respect to our early investors and investors.

> Since you have multiple proposals, it raises the question whether you would be able to deliver on both of them if they both get approved, especially since you say the core functionality of ASA Stats has the highest priority.

Yes, we'd manage to finish them. I'd be delighted if the core functionality makes us late, but unfortunately, there aren't so many new projects that arrive in the ecosystem that will spend too much of our time. I have to remind you that we had dealt with ten or twenty times more pressure the first year and we successfully deployed everything.

**Comment by ShaperOfEntropy** [comment]
Thank you very much for addressing my points!
This is reassuring and making the task for xGovs easier.
I wasn't aware that you operate that transparently. This is really to be commended!
I'm looking forward to following your project more closely.

**Pull request merged**